test/openssl/test_pkey_rsa.rb

```diff
@@ -46,63 +60,76 @@ def test_private
   end
 
   def test_new
-    key = OpenSSL::PKey::RSA.new(512)
-    assert_equal 512, key.n.num_bits
+    key = OpenSSL::PKey::RSA.new(2048)
+    assert_equal 2048, key.n.num_bits
     assert_equal 65537, key.e
     assert_not_nil key.d
 
-    # Specify public exponent
-    key2 = OpenSSL::PKey::RSA.new(512, 3)
-    assert_equal 512, key2.n.num_bits
-    assert_equal 3, key2.e
+    key2 = OpenSSL::PKey::RSA.new(2048, 65537)
+    assert_equal 2048, key2.n.num_bits
+    assert_equal 65537, key2.e
```

Comment:
The `RSA.new(a, b)` form is tested in `test_new_exponent` only in non-FIPS case. Here, the `key2` test is tested in FIPS case too. Considering this fact, would you still like to remvoe the `key2` test in the `test_new`?

Reply:
ruby/openssl doesn't change any behaviors depending on whether the OpenSSL is running in the FIPS mode or not. Looking at the implementation (https://github.com/ruby/openssl/blob/13b03ba4ed429cc8d46f5643cf5f9dd40275b4b2/lib/openssl/pkey.rb#L343-L348), it would be very unlikely we can introduce a bug in ruby/openssl that breaks `RSA.new(2048, 65537)` but doesn't break `RSA.new(512, 3)`.

I don't think it has zero value, but given the drawbacks (slowing down the tests), I feel it's not worth it.

Reply:
All right. In my understanding with the above implementation, the `key` test `key = OpenSSL::PKey::RSA.new(2048)` is calling the `generate(size, exp = 0x10001, &blk)` setting the default value `0x10001` to the `exp`. So, the `OpenSSL::PKey::RSA.new(2048)` covers the case of the `OpenSSL::PKey::RSA.new(2048, 65537)` in FIPS.

I will remove the `key2` test from both `test_new` and `test_s_generate`.

Reply:
Fixed this on the latest rebase.